testing/sampletests/main.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The sampletests command adds the region tags tested by each test to the test
+// name. It reads from stdin and writes to stdout.
+//
+// For example:
+//     TestFoo -> TestFoo__foo_hello_world__foo_hello_gopher
+//
+// It only looks at direct function calls by tests, not transitive calls.
+//
+// There are some duplicate region tags, but they aren't tracked anywhere else,
+// so it's OK if they are "applied" to more than one test.
+//
+// There are some duplicate test names, which means it's possible to get names
+// like:
+//     TestCreate__spanner_create__bigtable_create
+// Again, this is an acceptable tradeoff. It only happens when both regions are
+// tested. If only one is tested, only one would show up. Failures are
+// attributed to both snippets, though.
+//
+// The test coverage over all regions is printed at the end.
+//
+// To get the number of unique region tags in the repo manually, run:
+//     grep -RoPh '\[START \K(.+)\]' | sort | uniq | wc -l
```

Comment:
you could also use `sort -u` in place of `sort | uniq` so that it's all done when sorting. For large datasets this tends to be faster.

Reply:
Done.